packages/exchange-io-erc1888/src/deposit-watcher/deposit-watcher.service.ts

```diff
@@ -107,8 +107,8 @@ export class DepositWatcherService implements OnModuleInit {
 
         const deposit: CreateDepositDTO = {
             transactionHash,
-            address: from as string,
-            amount: value as string,
+            address: from,
+            amount: value.toString(),
```

Comment:
It is safer to use `String(value)` (If `value` can be `null` or 'undefined`)

Reply:
Fair point, but the ethers lib always returns a `value`, of type ethers BigNumber.

Besides, I don't think `String(value)` would work properly fornull or undefined because the event expects a stringed number, not just any string.

<img width="569" alt="Screen Shot 2021-02-01 at 15 32 06" src="https://user-images.githubusercontent.com/9353642/106472187-bb03cc80-64a2-11eb-9929-2627ad102264.png">

Reply:
Good point though, we should add a check to not emit the event if the value is incorrect.

Reply:
Added.

Reply:
under the hood `String()` calls `.toString()` method if possible, so it is fine to use it all the time.

Reply:
But here `.toString()` is ok since we have validation above

Reply:
I would disagree on using `String()`. I would prefer to get an error when `value == undefined` than to convert `undefined` to `'undefined'`. When it errors it's at least simpler to debug than getting an unexpected string.